docs/markdown/recursor/dnssec.md

```diff
@@ -0,0 +1,116 @@
+As of 4.0.0, the PowerDNS Recursor has support for DNSSEC processing and
+experimental support for DNSSEC validation.
+
+# DNSSEC settings
+The PowerDNS Recursor has 5 different levels of DNSSEC processing, which can be
+set with the [`dnssec`](settings.md#dnssec) setting in the `recursor.conf`. In
+order from least to most processing, these are:
+
+## `off`
+In this mode, **no** DNSSEC processing takes place. The PowerDNS Recursor will
+not set the DNSSEC OK (DO) bit in the outgoing queries and will ignore the DO and
+AD bits in queries. In this mode, the behaviour is equal to the PowerDNS Recursor
+3.X.
+
+## `process-no-validate`
+The default mode. In this mode the Recursor acts as a "security aware, non-validating"
+nameserver, meaning it will set the DO-bit on outgoing queries and will provide
+DNSSEC related RRsets (NSEC, RRSIG) to clients that ask for it (by means of a
+DO-bit in the query). It will not do any validation in this mode, not even when
+requested by the client.
+
+## `process`
+When `dnssec` is set to `process` the behaviour is similar to [`process-no-validate`](#process-no-validate).
+However, when the query has the AD-bit set, the recursor will try to validate the
+data and set the AD-bit in the response when the data is validated and send a
+SERVFAIL on a bogus answer.
+
+## `validate-log`
+In this mode, the recursor will attempt to validate all data it retrieves from
+authoritative servers, regardless of the client's DNSSEC desires, and will log the
+validation result. This mode can be used to determine the extra load and amount
+of possibly bogus answers before turning on full-blown validation.
+
+## `validate`
+The highest mode of DNSSEC processing. In this mode, all queries will be be validated
+and will be answered with a SERVFAIL in case of bogus data, regardless of the
+client's request.
+
+## What, when?
+The descriptions above are a bit terse, here's a table describing different scenarios
+with regards to the `dnssec` mode.
+
+|    | `off` | `process-no-validate` | `process` | `validate-log` | `validate` |
+|:------------|:-------|:-------------|:-------------|:-------------|:-------------|
+|Perform validation| No | No | Only on +AD from client | Always (logs result) | Always |
+|SERVFAIL on bogus| No | No | Only on +AD from client | Only on +AD from client | Always |
+|AD in response on authenticated data| Never | Never | Only on +AD from client | Only on +AD from client | Only on +AD from client |
+|RRSIGs/NSECs in answer on +DO from client| No | Yes | Yes | Yes | Yes |
+
+# Trust Anchor Management
+The PowerDNS Recursor ships with the DNSSEC Root key built-in. However, all trust
+anchors are configurable.
+
+## Trust Anchors
+Configuring DNSSEC keys must be done in the [`lua-config-file`](settings.md#lua-config-file),
+using `addDS`. This function takes 2 arguments, the node in the DNS-tree and the
+data of the correspoding DS record. To e.g. add a trust anchor for the root and
```

Comment:
corresponding. Do you have an i/aspell button in your editor? I find it is very useful.